amqp/amqp.go

```diff
@@ -626,18 +626,49 @@ func fromDelivery(d *amqp.Delivery) (*queue.Job, error) {
 	j.Raw = d.Body
 
 	if retries, ok := d.Headers[DefaultConfiguration.RetriesHeader]; ok {
-		retries, ok := retries.(int32)
-		if !ok {
-			return nil, ErrRetrievingHeader.New(DefaultConfiguration.RetriesHeader, d.MessageId)
-		}
+		switch r := retries.(type) {
+		case int16:
+			j.Retries = int32(r)
+
+		case int32:
+			j.Retries = int32(r)
+
+		case int64:
+			j.Retries = int32(r)
```

Comment:
it won't work! you'll have overflow issue (see: https://play.golang.org/p/-1sXpyO-KCY)
for in64 just do `.retries.(int32)` or modulo, or check if the value is < `math.MaxInt32`

Reply:
Retries in the job is an int32 so it cannot be bigger than that. This is done because the migration software converts all header integers to int64 but we don't expect them to be bigger that max_int32.

Reply:
I'll add the check just in case.

Reply:
ok, makes sense. you can also just in case do sth. like `int32(r) & math.MaxInt32`